src/modalities/evaluator.py

```diff
@@ -120,6 +77,51 @@ def evaluate(
         # )
         return result_dict
 
+    def _process_batches_in_data_loader(
+        self,
+        model: NNModel,
+        data_loader: LLMDataLoader,
+        global_train_sample_id: int,
+        local_sample_id_to_global_sample_id: Callable[[int], int],
+    ) -> Tuple[Dict[str, torch.Tensor], Dict[str, torch.Tensor]]:
+        losses = [f.create(self.local_rank) for f in self._loss_factories]
```

Comment:
not a big fan of the factories here, as they seem a bit overengineered. 
How about we instantiate the aggregated measures already in the hierarchical instantiation and pass the measures to the evaluator in main?
We could implement a `reset` function for all measures that clears out their internal state. In this case, we could get rid of the factories here, right?